### PR TITLE
Resource request

### PR DESCRIPTION
**Pull request by khaletuanncc**
## Release Notes

- **New Features**
	- Enhanced user interface for the resource request management system, including a new button for creating resource requests and improved search functionality.
	- Added new filtering options for bill accounts and refined layout for better usability.

- **Improvements**
	- Updated handling of null values for resource request properties, improving data representation.
	- Refactored methods for clarity and better error handling in resource request operations.
	- Introduced a new data transfer object for resource requests to enhance data management.

- **Bug Fixes**
	- Improved logic for updating billing information and ensuring accurate state reflection in the component.

Xử lí 
- khi add cv vào request, nếu request chưa có bill acc -> add bill acc cho request
- Sửa lỗi cập nhật cv infor không thành công ở resource request
- Migration ràng buộc khóa ngoại ở table ResourceRequest và thay đổi kiểu dữ liệu byte cho enum ProjectUserWorkingType

### CODE REVIEW

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/form-resourceRequestCVUser/form-resourceRequestCVUser.component.ts [hunk 1]
```diff
@@ -50,16 +51,26 @@ export class FormResourceRequestCVUserComponent extends AppComponentBase impleme
...
+    resourceCv.interviewDate = this.resourceRequestCV.interviewDate;
+    resourceCv.cvStatusId = this.resourceRequestCV.cvStatusId;
+    this.resourceRequestService.updateResourceRequestCV(resourceCv).subscribe((res: any) => {
+      if (res.success) {
         abp.notify.success("Update successfully")
-        this.dialogRef.close({ type: 'update', data: res.result})
+        this.dialogRef.close({ type: 'update', data: res.result })
       }
-      else{
+      else {
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/form-resourceRequestCVUser/form-resourceRequestCVUser.component.ts]: ok

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/form-resourceRequestCVUser/form-resourceRequestCVUser.component.ts [hunk 2]
```diff
@@ -36,10 +37,10 @@ export class FormResourceRequestCVUserComponent extends AppComponentBase impleme
...
   ngOnInit(): void {
     this.billInfoPlan = {
-                         resourceRequestId: this.input.resourceRequestId,
-                         userId : this.input.billUserInfo ? this.input.resourceRequestCV.userId : undefined,
-                         cvName: this.input.cvName,
-                         };
+      resourceRequestId: this.input.resourceRequestId,
+      userId: this.input.billUserInfo ? this.input.resourceRequestCV.userId : undefined,
+      cvName: this.input.cvName,
+    };
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/form-resourceRequestCVUser/form-resourceRequestCVUser.component.ts]: ok

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/form-resource-requestCV/form-resource-requestCV.component.ts [hunk 3]
```diff
@@ -134,22 +134,20 @@ export class ResourceRequestCVComponent extends AppComponentBase implements OnIn
...
+    this.dialogRef.close(createResourceRequestCV);
+    }, () => this.isLoading = false)
+  } else {
+    let createResourceRequestCV = this.formatInterviewDateToVN(resourceCv);
+    this.resourceRequestService.updateResourceRequestCV(createResourceRequestCV).pipe(catchError(this.resourceRequestService.handleError)).subscribe((res) => {
       abp.notify.success("Update Successfully!");
       this.dialogRef.close(res.result);
     }, () => this.isLoading = false)
-   }
+  }
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/form-resource-requestCV/form-resource-requestCV.component.ts]: ok

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts [hunk 4]
```diff
@@ -499,6 +499,18 @@ export class RequestResourceTabComponent
...
+            cvName: rs.cvName,
+            resourceRequestId: rs.resourceRequestId,
+            userId: rs.userId
+          }
+          this.resourceRequestService.UpdateBillInfoPlan(req).pipe(catchError(this.resourceRequestService.handleError)).subscribe(data => {
+            const res = (data as any).result;
+            request.billUserInfo = res.billUserInfo;
+            request.cvName = res.cvName
+          });
+        }
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts]: ok

File: angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts [hunk 5]
```diff
@@ -499,6 +499,25 @@ export class RequestResourceTabComponent
...
+              next: (data: { result: { billUserInfo: any; cvName: string } }) => {
+                if(data?.result) {
+                  request.billUserInfo = data.result.billUserInfo;
+                  request.cvName = data.result.cvName;
+                }
+              },
+              error: () => {},
+              complete: () => {}
+            });
+        }
```
[the remark this answers is not in the document]
Comment (khaletuanncc) [reply angular/src/app/modules/delivery-management/delivery/request-resource-tab/request-resource-tab.component.ts:520]: unnecessary

**Review by tiennguyenhuuncc — APPROVED** [review]

**Pull request merged**